src/display/canvas.js

```diff
@@ -218,7 +212,7 @@ var CachedCanvases = (function CachedCanvasesClosure() {
         // reset canvas transform for emulated mozCurrentTransform, if needed
         canvasEntry.context.setTransform(1, 0, 0, 1, 0, 0);
       } else {
-        var canvas = createScratchCanvas(width, height);
+        var canvas = CanvasGraphics.DOMcanvasFactory.create(width, height);
```

Comment:
I only ran gulp unittest and lint, but forgot to test the viewer. Thanks for pointing out, i will take care of that.